thali/thalireplicationmanager.js

```diff
@@ -33,30 +43,93 @@ ThaliReplicationManager.events = {
   START_ERROR: 'startError',
   STOP_ERROR: 'stopError',
   CONNECT_ERROR: 'connectError',
-  DISCONNECT_ERROR: 'disconnectError',
-  SYNC_ERROR: 'syncRetry'
+  DISCONNECT_ERROR: 'disconnectError'
 };
 
 /**
-* Starts the Thali replication manager with the given device name and port number
-* @param {String} deviceName the device name to advertise.
+* Returns the current device's id if available. If not, the provided callback
+* function is saved and called when the id does become available.
+* @param {Function} cb the callback which returns the current device's id.
+*/
+ThaliReplicationManager.prototype.getDeviceIdentity = function (cb) {
+  if(this._deviceIdentityFlag == deviceIdentityFlag.deviceIdentityAvailable) {
+    cb(null, this._deviceName);
+    return;
+  }
+  if(this._deviceIdentityFlag == deviceIdentityFlag.noDeviceIdentitySet) {
+    this._deviceIdentityFlag = deviceIdentityFlag.gettingDeviceIdentity;
+    getPublicKeyHash.call(this, cb);
+    return;
+  }
+  if(this._deviceIdentityFlag == deviceIdentityFlag.gettingDeviceIdentity) {
+    // save the callback for future
+    this._deviceIdentityListeners.push(cb);
+  }
+}
+
+function getPublicKeyHash(cb) {
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
```

Comment:
I fixed #2 & #3. But I do not understand #1. I am already adding cb to _deviceIdentityListeners at line 66. Please clarify.